File: app/src/main/java/com/app/id/dagger/Injector.java
package com.app.id.dagger;

import android.content.Context;

public abstract class Injector {

    private static final String APP_COMPONENT_SERVICE = "com.app.id.AppComponent";
    private static final String ACTIVITY_COMPONENT_SERVICE = "com.app.id.dagger.ActivityComponent";

    @SuppressWarnings({"WrongConstant"})
    public static AppComponent obtainAppComponent(Context context) {
        return (AppComponent) context.getApplicationContext().getSystemService(APP_COMPONENT_SERVICE);
    }

    @SuppressWarnings("WrongConstant")
    public static ActivityComponent obtainActivityComponent(Context context) {
        return (ActivityComponent) context.getSystemService(ACTIVITY_COMPONENT_SERVICE);
    }

    public static boolean matchesAppComponentService(String name) {
        return APP_COMPONENT_SERVICE.equals(name);
    }

    public static boolean matchesActivityComponentService(String name) {
        return ACTIVITY_COMPONENT_SERVICE.equals(name);
    }

    private Injector() {
        throw new AssertionError("No instances.");
    }
}

File: app/src/main/java/com/app/id/dagger/Injector.kt
package com.app.id.dagger

import android.content.Context

abstract class Injector {

    private constructor()

    init {
        throw AssertionError("No instances.")
    }

    companion object {

        private val APP_COMPONENT_SERVICE = "com.app.id.AppComponent"
        private val ACTIVITY_COMPONENT_SERVICE = "com.app.id.dagger.ActivityComponent"

        @SuppressWarnings("WrongConstant")
        fun obtainAppComponent(context: Context): AppComponent {
            return context.applicationContext.getSystemService(APP_COMPONENT_SERVICE) as AppComponent
        }

        @SuppressWarnings("WrongConstant")
        fun obtainActivityComponent(context: Context) =
                context.getSystemService(ACTIVITY_COMPONENT_SERVICE) as ActivityComponent

        fun matchesAppComponentService(name: String) = APP_COMPONENT_SERVICE == name

        fun matchesActivityComponentService(name: String) = ACTIVITY_COMPONENT_SERVICE == name
    }
}
